compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -472,10 +471,18 @@ pub(crate) unsafe fn optimize_with_new_llvm_pass_manager(
         sanitizer_options.as_ref(),
         pgo_gen_path.as_ref().map_or(std::ptr::null(), |s| s.as_ptr()),
         pgo_use_path.as_ref().map_or(std::ptr::null(), |s| s.as_ptr()),
+        config.instrument_coverage,
+        config.instrument_gcov,
         llvm_selfprofiler,
         selfprofile_before_pass_callback,
         selfprofile_after_pass_callback,
+        extra_passes.as_ptr().cast(),
+        extra_passes.len(),
     );
+
+    if let Some(err) = llvm::last_error() {
+        cgcx.create_diag_handler().warn(&format!("Invalid -Cpasses: {}", err));
```

Comment:
Accept a `diag_handler` in this function like `optimize` does?

Reply:
Additionally, this isn't obvious to me that the error from the call above will always be about `-Cpasses`. It would make more sense to me to do this formatting on the cpp side so that this code is more future-proof.